dependency-convergence/src/main/java/ArtifactData.java

```
@@ -0,0 +1,263 @@
+      metadataFile.deleteOnExit();
+
+      URL url = new URL(metadataUrl);
+      FileUtils.copyURLToFile(url, metadataFile);
+
+      MetadataXpp3Reader reader = new MetadataXpp3Reader();
+      Metadata metadata = reader.read(new FileInputStream(metadataFile));
+
+      if (metadata.getVersioning() == null) {
+        return new LatestMetadata("", "");
+      }
+
+      String latestVersion;
+      if (metadata.getVersioning().getLatest() == null) {
+        latestVersion = "";
+      } else {
+        latestVersion = metadata.getVersioning().getLatest();
+      }
+
+      String lastUpdated = metadata.getVersioning().getLastUpdated();
+      if (lastUpdated != null && !lastUpdated.isEmpty()) {
+        Date date = dateFormat.parse(lastUpdated);
+        lastUpdated = outputFormat.format(date);
+      } else {
+        lastUpdated = "";
+      }
+      return new LatestMetadata(latestVersion, lastUpdated);
+    } catch (XmlPullParserException | ParseException | IOException ignored) {
+      return new LatestMetadata("", "");
+    }
+  }
+
+  /**
+   * Grabs the Github URL from the POM's scm section. If the URL is not found, then this checks the
+   * associated parent POM (given that the current POM is not already the parent of our artifact)
+   *
+   * @param pomUrl the Maven URL of the artifact's POM file
+   * @return Artifact's Github URL
+   */
+  private static String getScmGithubUrl(String pomUrl) {
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      BufferedInputStream input = new BufferedInputStream(url.openStream());
+      FileUtils.copyInputStreamToFile(input, pomFile);
+      MavenXpp3Reader read = new MavenXpp3Reader();
+      Model model = read.read(new FileInputStream(pomFile));
+      if (model == null) {
+        return null;
+      }
+      return model.getUrl();
+    } catch (XmlPullParserException | IOException ignored) {
+    }
+    return null;
```

move return into catch block; almost never is an empty catch block OK